src/org/mozilla/javascript/JavaMembers.java

```diff
@@ -918,3 +740,988 @@ public Object getDefaultValue(Class<?> hint)
     Field field;
     Object javaObject;
 }
+
+class JavaMembersNew extends JavaMembers
```

Comment:
There is code in this (very old) codebase that has more than one top-level Java class per .java source file, but that's not a great pattern and something that Java developers rarely do any more. I would prefer if the various classes in this file are either all nested classes (they can be "static" if they don't need access to the parent) or if they were in different source files. (And since this change basically looks like a total rewrite, it might be easier to see what changed.)